contrib/vmallocinfo.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env drgn
+# Copyright (c) Canonical Ltd.
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+""" Script to dump vmallocinfo status using drgn"""
+
+import re
+from typing import Optional
+
+from drgn import IntegerLike, Program
+from drgn.helpers.linux.mm import for_each_vmap_area
+
+VMAP_RAM = 0x1  # indicates vm_map_ram area
+
+VM_IOREMAP = 0x00000001  # ioremap() and friends
+VM_ALLOC = 0x00000002  # vmalloc()
+VM_MAP = 0x00000004  # vmap()ed pages
+VM_USERMAP = 0x00000008  # suitable for remap_vmalloc_range
+VM_DMA_COHERENT = 0x00000010  # dma_alloc_coherent
+VM_SPARSE = 0x00001000  # sparse vm_area. not all pages are present.
+
+
+def is_vmalloc_addr(prog: Program, addr: IntegerLike) -> Optional[bool]:
+    vmcoreinfo = prog["VMCOREINFO"].string_()
+    match = re.search(
+        rb"^NUMBER\(VMALLOC_START\)=(0x[0-9a-f]+)$", vmcoreinfo, flags=re.M
+    )
+    if match:
+        VMALLOC_START = int(match.group(1), 16)
+
+    match = re.search(rb"^NUMBER\(VMALLOC_END\)=(0x[0-9a-f]+)$", vmcoreinfo, flags=re.M)
+    if match:
+        VMALLOC_END = int(match.group(1), 16)
+
+    try:
+        return True if addr >= VMALLOC_START and addr < VMALLOC_END else False
+    except:
+        return None
+
+
+for vmap_area in for_each_vmap_area(prog):
+    if not vmap_area.vm:
+        if vmap_area.flags & VMAP_RAM:
+            print(
+                f"0x{vmap_area.va_start:x}-0x{vmap_area.va_end:x} {vmap_area.va_end - vmap_area.va_start:10d} vm_map_ram"
+            )
+            continue
+    v = vmap_area.vm
+    print(
+        f"0x{v.addr.value_():x}-0x{(v.addr+v.size).value_():x} {v.size.value_():10d}",
+        end="",
+    )
+    if v.caller:
+        print(f" {prog.symbol(v.caller.value_()).name}", end="")
```

Comment:
I ran this on a system where I didn't have debug info for modules, and the symbol lookup failed with a `LookupError`. It would be nice to catch the `LookupError` and fall back to printing the raw hex address.

Other than that, these scripts are great, thank you!

Reply:
I updated patches.
Thank you for your advice.